### PR TITLE
v1.10.4 audit + v1.10.5 delta case drop + v1.10.6 F21 isomorph fixes

### PR DESCRIPTION
**Pull request by NeverMore93**
## Summary

Two release commits merging `dev_main_20260506` into `main`:

- **v1.10.4 (`1b59d98`)** — Audit scope expanded to user-visible output; removed rendering-layer plugin-authored labels (e.g. \"**结论**\"). Title promoted to immutable CardKit `header.title`.
- **v1.10.5 (`7b00548`)** — Architecture cleanup + reasoning field bug fix:
  - Dropped entire `message.part.delta` case in event.ts (30+ lines of dead-emit + fallback-only code).
  - Eliminated reasoning field pollution: when models like GPT-5/Claude-thinking emit `reasoning_content`/`reasoning_details` deltas with interleaved reasoning, the prior `(event as any)` strip dropped the `field` discriminator, leaking reasoning stream into the agent main reply textBuffer. Removing the case eliminates the bug entirely.
  - Removed dead `setTitle()` public method (silent no-op since v1.10.4; 0 callers in src/).
  - Removed duplicate `normalizeBlockMarkdown` in `createReplyCardView`.
  - Extracted `FORM_SUBMIT_RECEIVED_TOAST` (P3) and `FORM_SUBMIT_PROCESSING_TOAST` (P1) constants — two distinct toast semantics shared between `gateway.ts` and `interactive.ts`.

### Architecture docs

- New `CLAUDE.reply-rendering.md` (701 lines) — runtime data flow architecture with field-level pipeline diagram, type reference, and TODO list.
- Refreshed `prompts/AUDIT.md` — prompt injection point audit reflecting rendering-layer changes.
- Synced `src/feishu/CLAUDE.md`, `src/handler/CLAUDE.md`, `prompts/CLAUDE.md`.

### Trade-off accepted

Fallback path (`mirrorTextToMessage=true`, only triggers on CardKit failure) degrades from per-character typing to per-part-snapshot block refresh. Acceptable since fallback is an edge case — users get correct full content, just less smooth typing animation during the rare CardKit outage.

## Test plan

- [x] `npm run typecheck` clean
- [x] `npm run build` produces 182.84 KB bundle (~1KB smaller than v1.10.4 due to delta case removal)
- [ ] Manual: send normal message in Feishu p2p chat → verify card streams smoothly via `message.part.updated` snapshots (no regression in main path)
- [ ] Manual: same in group chat with @mention → ensure structured card with header/status/text/details works
- [ ] Manual: trigger `feishu_send_card` form interaction → verify P3 toast \"📋 已收到提交\" and P1 toast \"📋 已收到，正在处理...\"
- [ ] Manual: `permission.asked` / `question.asked` paths still produce interactive cards
- [ ] Manual: abort button still terminates run with state `aborted`
- [ ] Manual (edge case): force CardKit failure (block API) → verify fallback to plain text message still updates content
- [ ] Manual (model with reasoning): use a model that emits interleaved reasoning → verify main reply card shows only main reply text, not reasoning content

## Summary by CodeRabbit

* **Bug Fixes**
  * Unified toast phrasing to neutral “received / processing” feedback to avoid misleading success messages.
  * Removed confusing “conclusion” label from reply cards; replies now surface as cleaner reply text.

* **Documentation**
  * Added comprehensive prompt-injection audit and governance guidance.
  * Expanded runtime reply-streaming architecture and type/reference docs.
  * Established testing and fixture guidelines for repository tests.

* **Chores**
  * Version bumped to v1.10.6.

### CODE REVIEW

**Comment by NeverMore93** [comment]
## v1.10.6 增量补充（commit 9a642a5）

修复 PR #74 在 v1.10.5 修 F21 send_message/form_submit 反模式时**漏扫的 3 个同构 case**——同样是"toast 完成态 + v2Client async fire-and-forget 失败仅 log"结构，但漏在 grep 文案的视野盲区里。

### 3 个 F21 同构反模式（`src/handler/interactive.ts`）

| 反模式 | 位置 | toast | async caller |
|---|---|---|---|
| permission_reply | `:587-591` | `"✅ 已允许" / "❌ 已拒绝"` (success/warning) | `void v2Client.permission.reply().catch(onReplyFailed)` (line 557-560) |
| question_reply | `:594-598` | `"✅ 已回答"` (success) | `void v2Client.question.reply().catch(onReplyFailed)` (line 562-565) |
| abort_reply | `:478-516, :600-602` | `buildToast("success", "已接收中断请求")` | `void v2Client.session.abort().catch(...)` (line 501-515) |

**用户欺骗证据**：toast 在飞书 3 秒回调窗口内立即返回完成态，但实际 async 转发可能因网络/v2Client 错误失败。失败时 `onReplyFailed` 仅 `emitPhase("error", failureBody)` 在折叠面板更新——**用户面无对等失败提示**，agent 仍在等待原请求。

### 修法（与 v1.10.5 send_message/form_submit 一致）

```diff
- toast: { type: isReject ? "warning" : "success",
-          content: isReject ? "❌ 已拒绝" : "✅ 已允许" }
+ toast: { type: "info",
+          content: isReject ? "🚫 已收到拒绝请求，正在转交..."
+                            : "📨 已收到允许请求，正在转交..." }
```

`type: "info"` + 进行态文案表达"已接收，正在处理"，与异步结果对齐——既不欺骗用户成功，也不报告失败（事实上 toast 时刻无法知道）。

### 方法论补救（已沉淀）

- **`src/handler/CLAUDE.md` 末尾新增"反模式修复回归原则"**：要求修反模式 PR 必须按**结构而非文案**扫描同构（grep `void deps.v2Client.*.then().catch(...)` 而非 `"已发送"`/`"已提交"`）
- **本地 `docs/fallback-design-rules.md § 11`** 加详细规则（docs/ 在 .gitignore 内不入 git，仅作 Claude review 必读输入）
- **inventory § 4.1 "反模式清零"声明被本次扫描证伪**：今后修反模式只写"已修 X / Y / Z 三处实例，结构同构扫描已完成"

### Test plan

- [x] typecheck 通过
- [x] build 成功（dist 182.97 KB，+130 字节符合预期）
- [ ] 手动：群聊触发 permission/question 卡片，点击按钮 → 验证 toast 文案为进行态
- [ ] 手动：长任务运行时点中止按钮 → 验证 toast 文案为 "已接收中断请求，正在停止回答" 而非 success 视觉
- [ ] 手动：模拟 v2Client 失败场景（网络断开等）→ 验证用户看到的不是矛盾信号

File: src/handler/event.ts [hunk 1]
```diff
@@ -281,6 +249,9 @@ async function handleMessagePartUpdated(
   }
 
   // message.part.updated 只处理 text 类型的全量快照（delta 已由 message.part.delta 处理）
+  // 已知忽略的 part type（OpenCode SDK 联合类型 12 个成员，其中 tool/reasoning 已上面分支处理）：
+  // subtask / file / step-start / step-finish / snapshot / patch / agent / retry / compaction
+  // 这些都是 agent 内部状态（步骤边界 / 文件操作 / 压缩等），目前无 use case 需投影到飞书卡片
```
[the remark this answers is not in the document]
Comment (NeverMore93) [reply src/handler/event.ts]: Fixed in 622e2c0. Updated stale inline comment at event.ts:251 to reflect v1.10.5 delta case removal: now reads "v1.10.5 起 delta 路径已移除，见本文件顶部 switch 注释" with explicit cross-ref to the removal comment block.

File: CLAUDE.reply-rendering.md [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+| T1 | delta `field` 未过滤 | event.ts:149-153 用 `as any` 强转，只取 `sessionID/messageID/delta`，丢弃 `field` 和 `partID`。插件已在 line 18-20 定义了含 `field` 的正确类型，但 handler 未使用 | 若 server 发非 text 字段的 delta（如 reasoning），会被错误拼接到 textBuffer | event.ts:155 后加 `if (props.field !== "text") break` |
+
+### P2 设计改进
+
+| # | 项 | 现状 | 影响 | 建议 |
+|---|---|------|------|------|
+| T2 | CardKit 失败一次即 degraded，无重试 | streaming-card.ts:303-310 的 `enqueue.catch(markDegraded)` 一次失败永久降级。网络抖动/飞书 API 429 等瞬态错误不应永久放弃 | 用户看到卡片"卡住"（UI 停在最后成功状态），agent 不知道 | enqueue catch 加 1-2 次重试（间隔 500ms），仍失败再 degraded |
+| T3 | delta 累积在正常路径下是白功 | event.ts:160 每秒 30-100 次 `textBuffer += delta`，但 text 消费走 `pollForResponse.onSnapshot`（整段替换）。delta 累积的 textBuffer 仅服务 mirrorTextToMessage 降级路径 | 正常路径下无效 CPU 开销 | 选项 A：加注释标注（零风险）；选项 B：降级触发时才累积（复杂）；选项 C：降级路径改用 snapshot（失去实时性） |
+| T4 | 200ms debounce 偏保守 | 注释引用 Vercel AI SDK 的 50ms，但选了 200ms（4 倍）。飞书 CardKit API 限流阈值远高于 5 次/秒 | 用户感知到的"打字速度"被人为压慢 | 实测飞书 API 限流后降到 100-150ms |
+| T5 | 两条 text 路径并存，降级反而更实时 | 正常路径：SSE → textBuffer → polling snapshot → 200ms debounce → CardKit；降级路径：SSE → textBuffer → sender.updateMessage（跳过 debounce） | 降级路径比正常路径更快显示文本，违反直觉 | 统一路径或标注设计意图 |
```
[the remark this answers is not in the document]
Comment (NeverMore93) [reply CLAUDE.reply-rendering.md]: Fixed in 622e2c0. Marked T1 / T3 (and T7 by extension) with strikethrough + ✅ 已修复 (v1.10.5) status, citing PR #74 / commit 7b00548. Also added § 11 top disclaimer noting that all event.ts:148-185 line refs and "delta vs updated" comparisons in the doc are v1.10.4 historical snapshots, with current SSE flow described inline. Future contributors will not be misled into patching non-existent code.

File: CLAUDE.reply-rendering.md [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+│  part.delta → textBuffer += delta        │                                          │                                       │
+│  → emit ProcessedAction                  │                                          │  整段替换（覆盖 delta 累积）             │
+│  { type:"text-updated", sessionId,       │                                          │  event.ts:288 注释：有意设计            │
+│    delta?, fullText? }                   │                                          └───────────────────────────────────────┘
+└──────────────┬───────────────────────────┘
+               │
+               │  ⚠️ chat.ts:446-449 text-updated case = NO-OP
+               │  text 不经 action-bus，走 polling snapshot 路径
+               │
+               ▼
```
[the remark this answers is not in the document]
Comment (NeverMore93) [reply CLAUDE.reply-rendering.md:88]: Partially fixed in 622e2c0. Added § 11 top disclaimer explicitly noting that § 1a / § 2 main diagram / § 4.5 delta-vs-updated comparison / § 6 fallback dual-path diagram are all v1.10.4 historical snapshots, with the current single-path SSE flow stated inline (message.part.updated → handleMessagePartUpdated → sender.updateMessage / pollForResponse.onSnapshot). Deeper rewrite of those individual sections (~50 lines, redrawing diagrams) is deferred to a separate doc-only follow-up PR. The disclaimer prevents misreading in the meantime; full diagram rewrite is tracked as backlog.
